pkg/proc/target_exec.go

```diff
@@ -586,10 +586,28 @@ func (grp *TargetGroup) StepInstruction() (err error) {
 	if ok, err := dbp.Valid(); !ok {
 		return err
 	}
-	err = grp.procgrp.StepInstruction(thread.ThreadID())
+	var isCall bool
+	instr, err := disassembleCurrentInstruction(dbp, thread, 0)
 	if err != nil {
 		return err
 	}
+	isCall = len(instr) > 0 && instr[0].IsCall()
+	if skipCalls && isCall {
+		// Step into call.
+		err = grp.procgrp.StepInstruction(thread.ThreadID())
+		if err != nil {
+			return err
+		}
+		err = grp.StepOut()
```

Comment:
I think the cleanup that's done at the end of the function should be before the call to StepOut (it's fixing the state of the current thread) and isn't needed afterwards (so this could just be `return grp.StepOut()`).